server/src/main/java/org/elasticsearch/search/aggregations/CardinalityUpperBound.java

```diff
@@ -84,4 +88,46 @@ public CardinalityUpperBound multiply(int bucketCount) {
      *   will collect per owning ordinal
      */
     public abstract CardinalityUpperBound multiply(int bucketCount);
+
+    /**
+     * Map the cardinality to a value. If the upper bound of the cardinality
+     * is unknown the call the first argument to produce the value, otherwise
+     * call the second argument with the estimated upper bound.
+     */
+    public abstract <R> R map(IntFunction<R> mapper);
```

Comment:
I think I could just have a method to return an `int` here, but in all non-test cases we'll want to do something like:

```
LongKeyedBucketOrds impl(Cardinality cardinality) {
  return cardinality.map(estimate -> {
    if (estimate < 2) return new FromSingle();
    if (estimate < Integer.MAX_VALUE) return new FromInts();
    return new FromMany();
  });
}
```

So `map` feels fairly natural. Also, it will stop compiling if we ever decide that we need to encode more stuff in the estimate.